samcli/lib/utils/colors.py

```diff
@@ -42,6 +48,9 @@ def __init__(self, colorize=True):
         colorize : bool
             Optional. Set this to True to turn on coloring. False will turn off coloring
         """
+        self.rich_logging = any(
```

Comment:
Non blocking for this one, somewhat related to my feedback in the original pr, being able to find if rich is enabled throughout execution would be valuable in places like this.